### PR TITLE
fixes #24496 - remove unattended helper

### PR DESCRIPTION
**Pull request by timogoebel**
This commit does the following:

- remove the unattended helper
- remove dead code in `grub_pass` and `ks_console` macros, see [Foreman Community](https://community.theforeman.org/t/grub-pass-ks-console/10610) for reasoning
- add tests for `grub_pass`, `root_pass` and `ks_console` macros
- rename `@subject` to `@scope` in tests, I believe the latter is a better name
- add a test for nested snippet rendering

### CODE REVIEW

**Review by ik5 — COMMENTED** [review]
For me at least it is hard to do a review without understanding the full context of the change.

Can you please explain (at the bug report or here) why that helper was removed?

Thank you :)

**Comment by timogoebel** [comment]
This is just a refactoring. A helper suggests, this has something todo with views. But it basically just adds macros for rendering. That‘s why I moved this to the proper place of the renderer.

File: lib/foreman/renderer/scope/macros/host_template.rb [hunk 1]
```diff
@@ -42,6 +42,19 @@ def host_param_false?(name)
             host.params.has_key?(name) && Foreman::Cast.to_bool(host.params[name]) == false
           end
 
+          def root_pass
+            check_host
+            host.root_pass
+          end
+
+          def grub_pass
+            ''
```
Comment (ares) [inline lib/foreman/renderer/scope/macros/host_template.rb]: So based on https://community.theforeman.org/t/grub-pass-ks-console/10610/3 should we add a deprecation warning before we entirely drop the content? if not and the removal is suggested, I'd first drop these macros from community-templates repo, sync them and the remove them completely.

**Comment by timogoebel** [comment]
I have added the functionality, so this commit is just refactoring + added tests (and safe to merge imho).

@ares, @ik5, @tbrisker: Does anyone of you want to hit the button?

**Comment by tbrisker** [comment]
[test foreman]

**Comment by timogoebel** [comment]
Rebased.

**Comment by timogoebel** [comment]
[test upgrade]
[test foreman]

**Comment by timogoebel** [comment]
@tbrisker: Tests passed. 🍏

**Review by tbrisker — APPROVED** [review]
Thanks @timogoebel !

**Pull request merged**